fuzz/fuzz_targets/kernel_primitives.rs

```diff
@@ -0,0 +1,144 @@
+#![no_main]
+
+use arbitrary::Arbitrary;
+use bitcoinkernel::{prelude::*, Block, Transaction};
+use libfuzzer_sys::fuzz_target;
+
+#[derive(Debug, Arbitrary)]
+enum FuzzOperation {
+    ParseBlock { data: Vec<u8> },
+    ParseTransaction { data: Vec<u8> },
+    CompareBlockTxExtraction { block_data: Vec<u8>, tx_index: u16 },
+}
+
+fuzz_target!(|op: FuzzOperation| {
+    match op {
+        FuzzOperation::ParseBlock { data } => {
+            let Ok(block) = Block::try_from(data.as_slice()) else {
+                return;
+            };
+
+            let tx_count = block.transaction_count();
+            assert_eq!(tx_count, block.transactions().count());
+
+            for (i, tx) in block.transactions().enumerate().take(1000) {
+                if let Ok(indexed_tx) = block.transaction(i) {
+                    assert_eq!(tx.txid(), indexed_tx.txid());
+                }
+
+                assert_eq!(tx.inputs().count(), tx.input_count());
+                assert_eq!(tx.outputs().count(), tx.output_count());
+
+                for (j, input) in tx.inputs().enumerate().take(10) {
+                    if let Ok(indexed_input) = tx.input(j) {
+                        let op1 = input.outpoint();
+                        let op2 = indexed_input.outpoint();
+                        assert_eq!(op1.txid(), op2.txid());
+                        assert_eq!(op1.index(), op2.index());
+                        assert_eq!(op1.is_null(), op2.is_null());
+                    }
+                }
+
+                for (j, output) in tx.outputs().enumerate().take(10) {
+                    if let Ok(indexed_output) = tx.output(j) {
+                        assert_eq!(output.value(), indexed_output.value());
+                        assert_eq!(
+                            output.script_pubkey().to_bytes(),
+                            indexed_output.script_pubkey().to_bytes()
+                        );
+                    }
+                }
+            }
+        }
```

Comment:
I think it would be nicer to extend the existing roundtrip targets with these additional tests. Both just consume a byte string and it would be good to have both of them consume a robust corpus of both hand-rolled block data and fuzzer inputs.

Reply:
Makes sense!